### ISSUE TITLE
unknown issue (can't figgure out whats causing it)

### ISSUE THREAD

**Issue by kronikpillow** [bakkeby/dwm-flexipatch#21]
so, this morning I woke up, and my Browser was on all 9 tags on its own, while none of the tags were toggled ... the browser was supposed to sit on its own on tag 2, but it was present on tag 1 2 3 4 5 6 7 8 9 while tag 2 wasn't tagged (I hope I'm explaining this correctly)

this bug is present since I discovered flexipatch, although I'm not sure which patch is causing it, I suspect it is the switchtag patch as I have been using it since day 1, but not entirely sure ...

many times before I had similar situations, where a window on 1 tag was present on 2+ more tags while I didn't toggle it to be present, and if I close that window on any of the tags, it gets closed on all tags, first i taught it is a error on my side, but just today i realized it was a bug since I just left my browser last night on tag 2, and it was on all 9 today

basically it is as if i have toggled a window to be sticky, while it's not sticky, I'm not even using the sticky patch

**Comment by bakkeby** [comment]
dwm does not do anything on it's own without input, so it's unlikely that it would just do it on it's own.

Potentially you might have some overlapping keyboard shortcuts / combinations. Could you attach the `config.h` and `patches.h` for your build?

**Comment by kronikpillow** [comment]
it's definetly the switchtag patch, here is a video showing what im trying to explain : [video](https://www.youtube.com/watch?v=zPd50NbI0DA) ... Im saying it's the switchtag patch because I have used my same build for 2 weeks without it, and this behavior wasn't present, then I rebuilt my dwm with the switchtag patch, and it started to happen again

my build / config file can be found on [kronikpillow/dwm-flexipatch](https://github.com/kronikpillow/dwm-flexipatch)

**Comment by bakkeby** [comment]
Ha, ok.

Have a look at:
https://dwm.suckless.org/customisation/tagmask/

For the tagmask rules configuration I see that you are using integer values, presumably expecting that the value of 2 should place the application on tag 2.
```
/* class               role        instance    title         tags mask    switchtag    iscentered   isfloating   monitor */
{ "Brave-browser",     NULL,       NULL,       NULL,         2,           2,           0,           0,           -1 },
```
the tag mask is a _binary_ representation of which tags the client ends up on. The binary value of 2 is 10, which means that it is incidentally placed on tag 2 in this case, but if you have the value of 3 for example then the binary value of that is 11, meaning the client is placed on both tag 1 and tag 2. If you use 9 then the client would be placed on tag 1 and 4.

OK, now for the switchtag rules I see that you also have a 2 here. The interpretation of this value is as follows:
```
This patch adds an extra configuration option for individual rules where:
   0 is default behaviour
   1 automatically moves you to the tag of the newly opened application and
   2 enables the tag of the newly opened application in addition to your existing enabled tags
```
E.g. let's say that you always have the browser on tag 9 (tagmask set to 1 << 8 as per the firefox example). If you don't use this patch, or have the rule set to 0, and you are on tag 1 when you start your browser, then that would open up on tag 9 "behind the scenes" so to say.

If you have the rule set to 1 then you'd also immediately switch to tag 9 after opening the browser.

And as per above, if you have it set to 2 then the tag is opened on the current tag in addition to the 9th tag. This might explain some of the gremlins in your system :D

**Comment by kronikpillow** [comment]
wait, yes If i understand It correctly, I have set it to spawn brave at tag 2, and to enable the tag which is the equivalent of mod+ctrl+2? ... but in the video the issue in question is Discord, and there is no Discord rule, and the tags in the video aren't enabled

**Comment by bakkeby** [comment]
So, if you were to change your configuration so that your brave browser starts up on tag 3 rather than tag 2, how would you go about doing that?

If there is no rule set up for Discord, why do you think that the switchtag patch would interfere with it?

If you exit or kill a program, then that program stops running. Why do you expect the program to keep living on another tag?

**Comment by kronikpillow** [comment]
I think im doing a bad job here of explaining what is happening ..
on the video at 1:20 i start shifting the view between tag 1 and 2, Discord is present on both Tag 1 & 2 like i have set the window to be sticky (I don't have the sticky patch installed),
when I switch the view from tag 1 to tag 2 Discord is present on Tag 2 but tag 1 isn't toggled to view, then when i switch view back to tag 1 Discord is present on tag 1 but tag 2 is not toggled... so what im saying here is that the tag where Discord is sitting originally (tag 1) is not toggled when i shift view to tag 2 yet Discord is there, like the window was sticky  

why would I think it has to do with the switchtag patch? well because this behavior is not present for me without it, I'v been now using for a while my build just without the switchtag patch, and there wasn't a single situation where I had this issue, while with the patch it keeps happening over and over again with basically any window at random times

**Comment by bashmarklets** [comment]
> im not entirely sure what this question has to do with anything

Perhaps you should consider looking into it :) I'm asking questions so that you can learn from it. If you had followed everything I have said up until now then you'd have figured it out by yourself.

Tags are not workspaces. Tags are, in my opinion, superior to workspaces.
You can say that Discord is on tag 1, Brave Browser is on tag 2, and your text editor is on tag 3. You can then decide to `view` tag 1, 2 and 3 together and Discord, Brave Browser and your text editor is shown on screen together.

Likewise, you can view tag 4, 5, and 6, and start a new application: calculator. The calculator application is now shown on tag 4, tag 5 and tag 6.

In your video, you start off with having Discord being displayed on tag 1 and tag 2. Chances are that both tags were viewed when you started Discord.

With your config as-is I believe you can easily replicate this by:
   - restart your computer
   - start Brave Browser
   - close Brave Browser
   - start Discord

This because in your configuration you have explicitly said that brave browser should start on tag 2 and for the switchtag option you have explicitly said that **it should enable that tag _in addition_ to your current tag**.

Changing the switchtag setting to 1 rather than 2 for the Brave Browser makes for a more intuitive behaviour as you then actually change to tag 2 rather than enabling tag 2 in addition to your current tag(s).

**Comment by bakkeby** [comment]
> im not entirely sure what this question has to do with anything

Perhaps you should consider looking into it :) I'm asking questions so that you can learn from it. If you had followed everything I have said up until now then you'd have figured it out by yourself.

Tags are not workspaces. Tags are, in my opinion, superior to workspaces.
You can say that Discord is on tag 1, Brave Browser is on tag 2, and your text editor is on tag 3. You can then decide to `view` tag 1, 2 and 3 together and Discord, Brave Browser and your text editor is shown on screen together.

Likewise, you can view tag 4, 5, and 6, and start a new application: calculator. The calculator application is now shown on tag 4, tag 5 and tag 6.

In your video, you start off with having Discord being displayed on tag 1 and tag 2. Chances are that both tags were viewed when you started Discord.

With your config as-is I believe you can easily replicate this by:
   - restart your computer
   - start Brave Browser
   - close Brave Browser
   - start Discord

This because in your configuration you have explicitly said that brave browser should start on tag 2 and for the switchtag option you have explicitly said that **it should enable that tag _in addition_ to your current tag**.

Changing the switchtag setting to 1 rather than 2 for the Brave Browser makes for a more intuitive behaviour as you then actually change to tag 2 rather than enabling tag 2 in addition to your current tag(s).

**Comment by kronikpillow** [comment]
Ok ty for your time :) will look into it
